packages/core/src/storage/StorageContext.ts

```diff
@@ -53,20 +52,19 @@ export async function storageContextFromDefaults({
       ));
     indexStore =
       indexStore || (await SimpleIndexStore.fromPersistDir(persistDir, fs));
-    vectorStore =
-      vectorStore || (await SimpleVectorStore.fromPersistDir(persistDir, fs));
-    imageVectorStore = storeImages
-      ? await SimpleVectorStore.fromPersistDir(
-          path.join(persistDir, DEFAULT_IMAGE_VECTOR_NAMESPACE),
+    if (!(ObjectType.TEXT in vectorStores)) {
+      vectorStores[ObjectType.TEXT] =
+        vectorStore ??
+        ((await SimpleVectorStore.fromPersistDir(
+          persistDir,
           fs,
-        )
-      : imageVectorStore;
+        )) as unknown as VectorStore);
```

Comment:
@himself65 this is not very nice. How can I make `SimpleVectorStore.fromPersistDir` return a `VectorStore` given the fact that it's build by the mixin (see source of SimpleVectorStore)?